### PR TITLE
Release 2.3.0

### PR DESCRIPTION
**Pull request by ggabernet**
Main changes:
- Template update to v2.5.1
- Improvements in lineage tree analysis

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/airrflow/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/airrflow _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules.json [hunk 1]
```diff
@@ -3,14 +3,20 @@
...
-            "custom/dumpsoftwareversions": {
-                "git_sha": "e745e167c1020928ef20ea1397b6b4d230681b4d"
-            },
-            "fastqc": {
-                "git_sha": "49b18b1639f4f7104187058866a8fab33332bdfe"
-            },
-            "multiqc": {
-                "git_sha": "49b18b1639f4f7104187058866a8fab33332bdfe"
+            "git_url": "https://github.com/nf-core/modules.git",
+            "modules": {
```
Comment (jfy133) [inline modules.json:7]: Is it correct there are only 3 modules?

File: modules/local/alakazam/alakazam_lineage.nf [hunk 2]
```diff
@@ -14,14 +14,14 @@ process ALAKAZAM_LINEAGE {
     tuple val(meta), path("${tab}"), emit: tab
     path "versions.yml" , emit: versions
     path("*.tsv")
-    path("Clone_tree_plots/*.pdf")
-    path("Graphml_trees/All_graphs_patient.graphml")
+    path("Clone_tree_plots/*.pdf"), emit: graph_plots optional true
+    path("Graphml_trees/*.graphml"), emit: graph_export optional true
```
Comment (jfy133) [inline modules/local/alakazam/alakazam_lineage.nf:18]:
```suggestion
    path("Clone_tree_plots/*.pdf"), emit: graph_plots, optional true
    path("Graphml_trees/*.graphml"), emit: graph_export, optional true
```

File: nextflow.config [hunk 3]
```diff
@@ -185,16 +207,14 @@ profiles {
...
+
 // Load igenomes.config if required
 if (!params.igenomes_ignore) {
     includeConfig 'conf/igenomes.config'
 } else {
     params.genomes = [:]
 }
 
-// Load modules.config for DSL2 module specific options
-includeConfig 'conf/modules.config'
```
Comment (jfy133) [inline nextflow.config:196]: Is there a reason to do this? You still have MultiQC and FastQC for example (or are they not used, and just left here for template-merge reasons?)

**Review by jfy133 — APPROVED** [review]
Didn't check the custom scripts but template changes LGTM

File: modules.json (hunk 1 above)
Comment (ggabernet) [reply modules.json:7]: yes, all other modules are local. Need to add them to `nf-core/modules` at some point but unlikely that other pipelines will use them as they are very TCR / BCR specific tools...

File: nextflow.config (hunk 3 above)
Comment (ggabernet) [reply nextflow.config:196]: the import is just moved to some lines below, to have it at the same spot as the template

Comment (jfy133) [reply nextflow.config:196]: Ahh missed that, :+1:

File: modules/local/alakazam/alakazam_lineage.nf [hunk 4]
```diff
@@ -14,14 +14,14 @@ process ALAKAZAM_LINEAGE {
     tuple val(meta), path("${tab}"), emit: tab
     path "versions.yml" , emit: versions
     path("*.tsv")
-    path("Clone_tree_plots/*.pdf")
-    path("Graphml_trees/All_graphs_patient.graphml")
+    path("Clone_tree_plots/*.pdf"), emit: graph_plots, optional true
+    path("Graphml_trees/*.graphml"), emit: graph_export, optional true
```
Comment (ggabernet) [inline modules/local/alakazam/alakazam_lineage.nf:18]:
```suggestion
    path("Clone_tree_plots/*.pdf"), emit: graph_plots optional true
    path("Graphml_trees/*.graphml"), emit: graph_export optional true
```

File: modules/local/alakazam/alakazam_lineage.nf (hunk 2 above)
Comment (ggabernet) [reply modules/local/alakazam/alakazam_lineage.nf:18]: seems like a comma there is not supported, just tested this

**Comment by ggabernet** [comment]
Thanks for your review @jfy133 !

**Review by ssnn-airr — APPROVED** [review]

**Comment by ggabernet** [comment]
Thank you @ssnn-airr

**Pull request merged**